File: app/src/main/java/net/kibotu/kalmanrx/app/MainActivity.java
package net.kibotu.kalmanrx.app;

import android.hardware.Sensor;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.common.android.utils.logging.Logger;

import net.kibotu.android.deviceinfo.library.Device;
import net.kibotu.kalmanrx.app.ui.AccelerationSensorFragment;
import net.kibotu.kalmanrx.app.ui.AccelerationSensorKalmanFragment;
import net.kibotu.kalmanrx.app.ui.AccelerationSensorLowPassFragment;

import java.util.List;

import static br.com.zbra.androidlinq.Linq.stream;
import static com.common.android.utils.extensions.FragmentExtensions.replace;
import static com.common.android.utils.extensions.FragmentExtensions.setFragmentContainerId;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<Sensor> sensorList = Device.getSensorList();
        for (Sensor sensor : sensorList)
            Logger.v(TAG, "Sensor: " + sensor.getName() + " [" + sensor.getType() + "]");

        Sensor accelerometer = stream(sensorList).firstOrNull(sensor
                -> sensor.getType() == android.hardware.Sensor.TYPE_ACCELEROMETER);

        if (accelerometer == null)
            return;

        replace(new AccelerationSensorFragment());

        setFragmentContainerId(R.id.fragment_container2);
        replace(new AccelerationSensorLowPassFragment());

        setFragmentContainerId(R.id.fragment_container3);
        replace(new AccelerationSensorKalmanFragment());
    }
}

File: app/src/main/java/net/kibotu/kalmanrx/app/MainActivity.kt
package net.kibotu.kalmanrx.app

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.exozet.android.core.utils.FragmentExtensions.replace
import com.exozet.android.core.utils.FragmentExtensions.setFragmentContainerId
import net.kibotu.android.deviceinfo.library.Device
import net.kibotu.kalmanrx.app.ui.AccelerometerSensorKalmanFragment
import net.kibotu.kalmanrx.app.ui.AccelerometerSensorLowPassFragment
import net.kibotu.kalmanrx.app.ui.AzimuthFragment
import net.kibotu.logger.LogcatLogger
import net.kibotu.logger.Logger
import net.kibotu.logger.Logger.logv


/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Logger.addLogger(LogcatLogger())

        val sensorList = Device.getSensorList()
        for (sensor in sensorList)
            logv("Sensor: ${sensor.name} [${sensor.type}]")

        // check if we have accelerometer
        val accelerometer = sensorList.firstOrNull { it.type == android.hardware.Sensor.TYPE_ACCELEROMETER } ?: return

        replace(AzimuthFragment())

        setFragmentContainerId(R.id.fragment_container2)
        replace(AccelerometerSensorLowPassFragment())

        setFragmentContainerId(R.id.fragment_container3)
        replace(AccelerometerSensorKalmanFragment())
    }
}
